lib/dynamoid/transaction/update.rb

```diff
@@ -0,0 +1,68 @@
+require_relative 'action'
+
+module Dynamoid
+  module Transaction
+    class Update < Action
+
+      def run_callbacks
+        (yield; return) unless model
+        model.run_callbacks(:save) do
+          model.run_callbacks(:update) do
+            model.run_callbacks(:validate) do
+              yield
+            end
+          end
+        end
+      end
+
+      def to_h
+        item = attributes.clone
+
+        # ensure types are correct for incoming attributes
+        all_types_string = item.keys.map{|k| [k,{type: :string}]}.to_h
+        item = add_timestamps(item, skip_created_at: true)
+        item = model.attributes.merge(item) if model
+        # TODO only update dirty attributes, deleting the inheritance field will not be needed
+        item.delete(model_class.inheritance_field) # do not update type field
+        item = Dynamoid::Dumping.dump_attributes(item, all_types_string.merge(model_class.attributes))
+
+        # set 'key' that is used to look up record for updating
+        hash_key = item.delete(model_class.hash_key)
+        key = {model_class.hash_key => hash_key}
+        key[model_class.range_key] = item.delete(model_class.range_key) if model_class.range_key?
+
+        # e.g. "SET #updated_at = :updated_at ADD record_count :i"
+        update_expression = options[:update_expression] || ("SET " + item.keys.map{ |k| "##{k} = :#{k}" }.join(", "))
```

Comment:
**suggestion:** it makes sense to use short generated placeholders like `a0`, `b1` etc to have expression string shorter. 

https://github.com/Dynamoid/dynamoid/blob/6d53f100f0478bc04c7c45646771182df603fdc8/lib/dynamoid/adapter_plugin/aws_sdk_v3/query.rb#L58-L59

Reply:
I don't mind the expression length in exchange for readability but I see the argument for brevity. I'll look into this.

Reply:
Shorter update expressions are now implemented.